### PR TITLE
Balder/wire shared executor into threading service.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst PR
No semantics, only wiring.

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/server/executorthreadingservice.h [hunk 1]
```diff
@@ -18,12 +18,14 @@ class ExecutorThreadingServiceStats;
...
 {
 private:
+    vespalib::ThreadStackExecutorBase & _sharedExecutor;
     vespalib::ThreadStackExecutor _masterExecutor;
     vespalib::BlockingThreadStackExecutor _indexExecutor;
     vespalib::BlockingThreadStackExecutor _summaryExecutor;
     ExecutorThreadService _masterService;
     ExecutorThreadService _indexService;
     ExecutorThreadService _summaryService;
+    ExecutorThreadService _sharedService;
```
Comment (geirst) [inline searchcore/src/vespa/searchcore/proton/server/executorthreadingservice.h]: ExecutorThreadService and the IThreadService interface is used to represent a single thread. This should not be used for the shared executor. I think we should just expose the shared executor as a vespalib::ThreadExecutor (or ThreadStackExecutorBase if needed).

**Review by geirst — CHANGES_REQUESTED** [review]
Overall looks good, please see comment regarding how shared executor is exposed from IThreadingService.

File: searchcore/src/vespa/searchcore/proton/server/executorthreadingservice.h (hunk 1 above)
Comment (baldersheim) [reply searchcore/src/vespa/searchcore/proton/server/executorthreadingservice.h]: Done

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**